src/sage/rings/species.py

```diff
@@ -0,0 +1,2255 @@
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: P = PolynomialSpecies(ZZ, ["X", "Y"])
+            sage: E4X = P(SymmetricGroup(4), {0: range(1, 5)}); E4X
+            E_4(X)
+            sage: E4Y = P(SymmetricGroup(4), {1: range(1, 5)}); E4Y
+            E_4(Y)
+            sage: P.degree_on_basis(E4X.support()[0])
+            4
+            sage: P.degree_on_basis(E4Y.support()[0])
+            4
+        """
+        return sum(m.grade())
+
+    @cached_method
+    def one_basis(self):
+        r"""
+        Returns SymmetricGroup(0), which indexes the one of this algebra,
+        as per :meth:`AlgebrasWithBasis.ParentMethods.one_basis`.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: P = PolynomialSpecies(ZZ, "X")
+            sage: P.one_basis()
+            1
+            sage: P2 = PolynomialSpecies(ZZ, "X, Y")
+            sage: P2.one_basis()
+            1
+        """
+        return self._indices.one()
+
+    @cached_method
+    def an_element(self):
+        """
+        Return an element of ``self``.
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: P = PolynomialSpecies(ZZ, "X")
+            sage: P.an_element()
+            1
+            sage: P2 = PolynomialSpecies(ZZ, "X, Y")
+            sage: P2.an_element()
+            1
+        """
+        return self.one()
+
+    def product_on_basis(self, H, K):
+        r"""
+        Return the product of the basis elements indexed by `H` and `K`.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: P = PolynomialSpecies(ZZ, "X")
+            sage: L1 = [P(H) for H in SymmetricGroup(3).conjugacy_classes_subgroups()]
+            sage: L2 = [P(H) for H in SymmetricGroup(2).conjugacy_classes_subgroups()]
+            sage: matrix([[F * G for F in L1] for G in L2])  # indirect doctest
+            [    X^5 X^3*E_2 X^2*C_3 X^2*E_3]
+            [X^3*E_2 X*E_2^2 E_2*C_3 E_2*E_3]
+
+        TESTS::
+
+            sage: P = PolynomialSpecies(ZZ, "X")
+            sage: X = P(SymmetricGroup(1))
+            sage: type(list(X^2)[0][1])
+            <class 'sage.rings.integer.Integer'>
+        """
+        return self._from_dict({H * K: ZZ(1)})
+
+    @cached_method
+    def powersum(self, s, n):
+        r"""
+        Return the combinatorial powersum species `P_n(X_s)`.
+
+        The species `P_n(X)` is introduced in [Labelle2008]_ as the
+        coefficient of `t^n/n` in `\log E(tX)`, where `E` is the
+        species of sets.
+
+        EXAMPLES::
+
+            sage: from sage.rings.species import PolynomialSpecies
+            sage: P = PolynomialSpecies(ZZ, "X")
+            sage: P.powersum(0, 4)
+            4*E_4 - 4*X*E_3 - 2*E_2^2 + 4*X^2*E_2 - X^4
+        """
+        assert n in ZZ and n > 0
```

Comment:
```suggestion
        if not (n in ZZ and n > 0):
            raise ValueError("n must be a positive integer")
```
Don't use `assert` for input checking.